frontend/src/metabase/admin/settings/components/SettingsSetting.jsx

```diff
@@ -33,62 +35,80 @@ const SETTING_WIDGET_MAP = {
-    setting: PropTypes.object.isRequired,
-    settingValues: PropTypes.object,
-    onChange: PropTypes.func.isRequired,
-    onChangeSetting: PropTypes.func,
-    autoFocus: PropTypes.bool,
-    disabled: PropTypes.bool,
-  };
+export const SettingsSetting = props => {
+  const { hash } = useLocation();
+  const [fancyStyle, setFancyStyle] = useState({});
+  const thisRef = useRef();
+
+  const { setting, settingValues, errorMessage } = props;
 
-  render() {
-    const { setting, settingValues, errorMessage } = this.props;
-    const settingId = settingToFormFieldId(setting);
+  useEffect(() => {
+    if (hash === `#${setting.key}` && thisRef.current) {
+      thisRef.current.scrollIntoView({
+        behavior: "smooth",
+        block: "center",
+      });
+      thisRef.current.focus();
 
-    let Widget = setting.widget || SETTING_WIDGET_MAP[setting.type];
-    if (!Widget) {
-      console.warn(
-        "No render method for setting type " +
-          setting.type +
-          ", defaulting to string input.",
-      );
-      Widget = SettingInput;
+      setFancyStyle({
+        background: alpha("brand", 0.1),
+        boxShadow: `0 0 0 1px ${color("brand")}`,
+      });
+
+      setTimeout(() => {
+        setFancyStyle({});
+      }, 1500);
     }
+  }, [hash, setting.key]);
 
-    const widgetProps = {
-      ...setting.getProps?.(setting, settingValues),
-      ...setting.props,
-      ...this.props,
-    };
+  const settingId = settingToFormFieldId(setting);
 
-    return (
-      // TODO - this formatting needs to be moved outside this component
-      <SettingRoot data-testid={`${setting.key}-setting`}>
-        {!setting.noHeader && (
-          <SettingHeader id={settingId} setting={setting} />
-        )}
-        <SettingContent>
-          {setting.is_env_setting && !setting.forceRenderWidget ? (
-            <SettingEnvVarMessage>
-              {jt`This has been set by the ${(
-                <ExternalLink href={getEnvVarDocsUrl(setting.env_name)}>
-                  {setting.env_name}
-                </ExternalLink>
-              )} environment variable.`}
-            </SettingEnvVarMessage>
-          ) : (
-            <Widget id={settingId} {...widgetProps} />
-          )}
-        </SettingContent>
-        {errorMessage && (
-          <SettingErrorMessage>{errorMessage}</SettingErrorMessage>
-        )}
-        {setting.warning && (
-          <SettingWarningMessage>{setting.warning}</SettingWarningMessage>
-        )}
-      </SettingRoot>
+  let Widget = setting.widget || SETTING_WIDGET_MAP[setting.type];
+  if (!Widget) {
+    console.warn(
+      "No render method for setting type " +
+        setting.type +
+        ", defaulting to string input.",
     );
+    Widget = SettingInput;
   }
-}
+
+  const widgetProps = {
+    ...setting.getProps?.(setting, settingValues),
+    ...setting.props,
+    ...props,
+  };
+
+  return (
+    // TODO - this formatting needs to be moved outside this component
```

Comment:
I think this still needs to move, right?

Reply:
Yeah, I left it there since I didn't really change anything. It's a TODO from 4 years ago though... Haha